plugins/vocabbuilder.koplugin/main.lua

```diff
@@ -63,6 +63,19 @@ local function saveSettings()
     G_reader_settings:saveSetting("vocabulary_builder", settings)
 end
 
+-- copied from readerhighlight.lua
+local function cleanupSelectedText(text)
+    -- Trim spaces and new lines at start and end
+    text = text:gsub("^[\n%s]*", "")
+    text = text:gsub("[\n%s]*$", "")
+    -- Trim spaces around newlines
+    text = text:gsub("%s*\n%s*", "\n")
+    -- Trim consecutive spaces (that would probably have collapsed
+    -- in rendered CreDocuments)
+    text = text:gsub("%s%s+", " ")
+    return text
+end
+
```

Comment:
Possibly worth making the ReaderHighlight one public and use it instead of duplicating code?

Reply:
Do I just add a line like `ReaderHighlight.cleanupSelectedText = cleanupSelectedText` or change the function to be under ReaderHighlight and also where it's being called?

Reply:
https://github.com/koreader/koreader/blob/802a9068705f4d1564fc24bd2412eb1dfdfaf994/frontend/languagesupport.lua#L153-L155

Reply:
Ah, good catch! Then, yeah, probably move it to util and update every call site ;).

Reply:
Do I do it or you guys do it? I'm not that confident in messing with the core codebase...

> Ah, good catch! Then, yeah, probably move it to util and update every call site ;).

Reply:
Done in https://github.com/koreader/koreader/pull/12477 ;).